### ISSUE TITLE
Drawing some circuits with MPL means they cannot be transpiled in parallel later

### ISSUE THREAD

**Issue by frankharkins** [Qiskit/qiskit-terra#3808]
### Information

- **Qiskit Terra version**: 0.12.0
- **Python version**: 3.7.4
- **Operating system**: OSX Catalina
(Jupyter Notebook)

### What is the current behavior?
Passing `execute()` a list of circuits gives `BrokenProcessPool` error if the circuit has been drawn with mpl _and_ has a reset in it.

### Steps to reproduce the problem
Create list of circuits with resets:
```
from qiskit import *
circuits = [QuantumCircuit(1) for i in range(2)]
for qc in circuits:
    qc.reset(0)
    qc.measure_all() 
```
Draw one using mpl:
`circuits[0].draw(output='mpl')`
Execute:
`job = execute(circuits, Aer.get_backend('qasm_simulator'))`

### What is the expected behavior?
Properly executed circuits.

**Comment by mtreinish** [comment]
I just tried this locally on linux (mostly out of curiosity) and was not able to reproduce this failure. This makes me think that this is related to https://github.com/Qiskit/qiskit-terra/pull/3491 which was a change for only macos in 0.12.0 as part of the 3.8 entablement. The one thing I'm still not sure about is how the reset gate is the trigger for the parallel execution to fail. Do you have the full traceback from the failure?

**Comment by frankharkins** [comment]
I agree it seems quite strange, but if I remove the resets or the mpl drawing the error disappears.

```
---------------------------------------------------------------------------
BrokenProcessPool                         Traceback (most recent call last)
<ipython-input-14-36defa612a53> in <module>
      1 circuits = code.get_circuit_list()
----> 2 job = execute(circuits, Aer.get_backend('qasm_simulator'))
      3 #results = job.result().get_counts('0')

/usr/local/anaconda3/lib/python3.7/site-packages/qiskit/execute.py in execute(experiments, backend, basis_gates, coupling_map, backend_properties, initial_layout, seed_transpiler, optimization_level, pass_manager, qobj_id, qobj_header, shots, memory, max_credits, seed_simulator, default_qubit_los, default_meas_los, schedule_los, meas_level, meas_return, memory_slots, memory_slot_size, rep_time, parameter_binds, **run_config)
    218                             optimization_level=optimization_level,
    219                             backend=backend,
--> 220                             pass_manager=pass_manager,
    221                             )
    222 

/usr/local/anaconda3/lib/python3.7/site-packages/qiskit/compiler/transpile.py in transpile(circuits, backend, basis_gates, coupling_map, backend_properties, initial_layout, seed_transpiler, optimization_level, pass_manager, callback, output_name)
    202                                   'in the coupling_map')
    203     # Transpile circuits in parallel
--> 204     circuits = parallel_map(_transpile_circuit, list(zip(circuits, transpile_configs)))
    205 
    206     if len(circuits) == 1:

/usr/local/anaconda3/lib/python3.7/site-packages/qiskit/tools/parallel.py in parallel_map(task, values, task_args, task_kwargs, num_processes)
    132             # Otherwise just reset parallel flag and error
    133             os.environ['QISKIT_IN_PARALLEL'] = 'FALSE'
--> 134             raise error
    135 
    136         Publisher().publish("terra.parallel.finish")

/usr/local/anaconda3/lib/python3.7/site-packages/qiskit/tools/parallel.py in parallel_map(task, values, task_args, task_kwargs, num_processes)
    122                 future = executor.map(_task_wrapper, param)
    123 
--> 124             results = list(future)
    125             Publisher().publish("terra.parallel.done", len(results))
    126 

/usr/local/anaconda3/lib/python3.7/concurrent/futures/process.py in _chain_from_iterable_of_lists(iterable)
    481     careful not to keep references to yielded objects.
    482     """
--> 483     for element in iterable:
    484         element.reverse()
    485         while element:

/usr/local/anaconda3/lib/python3.7/concurrent/futures/_base.py in result_iterator()
    596                     # Careful not to keep a reference to the popped future
    597                     if timeout is None:
--> 598                         yield fs.pop().result()
    599                     else:
    600                         yield fs.pop().result(end_time - time.monotonic())

/usr/local/anaconda3/lib/python3.7/concurrent/futures/_base.py in result(self, timeout)
    426                 raise CancelledError()
    427             elif self._state == FINISHED:
--> 428                 return self.__get_result()
    429 
    430             self._condition.wait(timeout)

/usr/local/anaconda3/lib/python3.7/concurrent/futures/_base.py in __get_result(self)
    382     def __get_result(self):
    383         if self._exception:
--> 384             raise self._exception
    385         else:
    386             return self._result

BrokenProcessPool: A process in the process pool was terminated abruptly while the future was running or pending.
```
Sorry it's a large one

**Comment by JamesSeaward** [comment]
I have just tried to reproduce on a Windows machine and also cannot reproduce it.

**Comment by nonhermitian** [comment]
This is an issue with Matplotlib that they have said they would fix in version 3.3.
